src/main/java/ch/njol/skript/expressions/ExprAllSignLines.java

```diff
@@ -0,0 +1,190 @@
+	private Expression<Block> blocks;
+	private boolean multiple = false;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		blocks = (Expression<Block>) exprs[exprs.length - 1];
+		multiple = !blocks.isSingle();
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected String[] get(Event e) {
+		Block[] blocks = this.blocks.getArray(e);
+		if (blocks.length == 0)
+			return new String[0];
+
+		List<String> lines = new ArrayList<>();
+
+		boolean isSignChangeEvent = e instanceof SignChangeEvent;
+		if (getTime() >= 0 && this.blocks.isDefault() && isSignChangeEvent && !Delay.isDelayed(e)) {
+			SignChangeEvent event = (SignChangeEvent) e;
+			lines.addAll(List.of(event.getLines()));
+		}
+
+		if (!multiple && (isSignChangeEvent && blocks[0].equals(((SignChangeEvent) e).getBlock())))
+			return lines.toArray(new String[0]);
+
+		for (Block block : blocks) {
+			if (!sign.isOfType(block))
+				continue;
+
+			Sign sign = (Sign) block.getState();
+			lines.addAll(Arrays.asList(sign.getLines()));
+		}
+
+		return lines.toArray(new String[0]);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.SET)
+			return new Class[]{String.class};
+		return null;
+	}
+
+	static boolean hasUpdateBooleanBoolean = true;
+
+	@SuppressWarnings("incomplete-switch")
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) throws UnsupportedOperationException {
+		Block[] blocks = this.blocks.getArray(e);
+		if (blocks.length == 0)
+			return;
+
+		for (Block block : blocks) {
+			if (!sign.isOfType(block))
+				continue;
+
+			if (e instanceof SignChangeEvent && block.equals(((SignChangeEvent) e).getBlock())) {
+				SignChangeEvent event = (SignChangeEvent) e;
+				if (getTime() >= 0 && !Delay.isDelayed(e)) {
+					switch (mode) {
+						case DELETE:
+							setAllLines(null, "", event);
+							break;
+						case SET:
+							assert delta != null;
+							setAllLines(null, (String) delta[0], event);
+							break;
+					}
+				}
+			}
+
+			Sign s = (Sign) block.getState();
+			switch (mode) {
+				case DELETE:
+					setAllLines(s, "", null);
+					break;
+				case SET:
+					assert delta != null;
+					setAllLines(s, (String) delta[0], null);
+					break;
+			}
+			if (hasUpdateBooleanBoolean) {
+				try {
+					s.update(false, false);
+				} catch (NoSuchMethodError err) {
+					hasUpdateBooleanBoolean = false;
+					s.update();
+				}
+			} else {
+				s.update();
+			}
+		}
+	}
+
+	private void setAllLines(@Nullable Sign sign, @NonNull String line, @Nullable SignChangeEvent event) {
+		for (int i = 0; i < 4; i++) {
+			if (event != null)
+				event.setLine(i, line);
+			else
+				sign.setLine(i, line);
+		}
+	}
+
+	@Override
+	public boolean setTime(int time) {
+		return super.setTime(time, SignChangeEvent.class, blocks);
```

Comment:
You're setting the time state to all here, but you're only supporting 0 and 1 aka present and future in the get. You need to only return if it's 0 or 1, or you need to add support for getting the former lines of the sign in the get method.

Example in the get method:
```java
protected String[] get(Event e) {
    //existing code...
    if (getTime() == -1 && isSignChangeEvent) {
        SignChangeEvent event = (SignChangeEvent) e;
        lines.addAll(List.of(((Sign)event.getBlock()).getLines()));
        return lines.toArray(String[]::new);
    }
}
```
```vb
if the former lines contains "[protected]":
    player does not have permission "admin.protected"
    message "&cYou don't have permission to edit this sign."
    cancel event
```